pyslackersweb/website/tasks.py

```diff
@@ -72,13 +72,15 @@ class Channel:
 
         repositories.sort(key=lambda r: r.stars, reverse=True)
 
-        await redis.set(cache_key, json.dumps([x.__dict__ for x in repositories[:6]]))
-
+        await redis.set(
+            cache_key, json.dumps([dataclasses.asdict(repo) for repo in repositories[:6]])
+        )
     except asyncio.CancelledError:
         logger.debug("Github cache refresh canceled")
-    except Exception:
+    except Exception:  # pylint: disable=broad-except
         logger.exception("Error refreshing GitHub cache")
-        raise
+
+    return repositories
```

Comment:
If we were to keep the return, I am not fond of swallowing the exception to return a possibly empty list. The caller wouldn't know if there was an error or if there are simply no repos. (same for below usages)

Reply:
We have to swallow the exception anyways as they are not handled by apscheduler. We would end up with asyncio warnings about unhandled exceptions.

Nobody is supposed to call those functions directly, they are handled by apscheduler and the `return` value is discarded anyways. IMO this allow us to tests those very easily

Reply:
I'll add a ticket to add an event handler for the exceptions: https://apscheduler.readthedocs.io/en/latest/userguide.html#scheduler-events

Reply:
#373